### ISSUE TITLE
`black` not respecting the local `pyproject.toml`

### ISSUE THREAD

**Issue by scallaway** [jose-elias-alvarez/null-ls.nvim#722]
Basically as per the title, I'm unable to format my files with `null_ls_module.builtins.formatting.black` based on the `pyproject.toml` file that is located in my project root directory.

I am, however, able to call `:! black %` and have it format respecting the `pyproject.toml` file.

The only other piece of information on this I could find (since there was nothing in this repo's issues or discussions) was this reddit post here; https://www.reddit.com/r/neovim/comments/s2o8ww/format_and_sort_on_save_with_lsp_and_nullls/ - this seems to be basically the same problem.

---

On a side note, I have a feeling that the same thing is happening with `isort`, since it reports that I have imports that are out of order, even _after_ I call `black` manually.

Is this because of the location in which the processes are spawned? And so since it's not in the directory it in theory _should_ be, it's not able to find the config file and is reverting back to the defaults?

---

I don't have the time at the moment to test with a minimal config, but I wanted to at least open this point to get any other voices on the matter.

**Comment by karljs** [comment]
This might be a duplicate of #700 where `black` just isn't running as expected at least some of the time, which I was able to reproduce there.

**Comment by TortoiseHive** [comment]
Same here

**Comment by jose-elias-alvarez** [comment]
I'm closing this issue because it lacks the information I need to investigate and act on it. Feel free to open a new issue with reproduction instructions if you're still having this issue.

**Comment by Jackevansevo** [comment]
For me this issue is fixed by the suggestion in this thread https://github.com/jose-elias-alvarez/null-ls.nvim/issues/848#issuecomment-1119040548


- https://github.com/jose-elias-alvarez/null-ls.nvim/issues/848

I.e. 


```lua
    require('null-ls').builtins.diagnostics.black.with {
      cwd = function(params)
        return vim.fn.fnamemodify(params.bufname, ':h')
      end,
    },
```

**Comment by scallaway** [comment]
> For me this issue is fixed by the suggestion in this thread [#848 (comment)](https://github.com/jose-elias-alvarez/null-ls.nvim/issues/848#issuecomment-1119040548)

Can confirm that this does indeed fix my issue!
